singlestoredb/management/job.py

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/env python
+"""SingleStoreDB Cloud Scheduled Notebook Job."""
+import datetime
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from enum import Enum
+
+from .utils import to_datetime
+from .manager import Manager
+
+
+
+class TargetType(Enum):
+    WORKSPACE = "Workspace"
+    CLUSTER = "Cluster"
+    VIRTUAL_WORKSPACE = "VirtualWorkspace"
+
+
+class Status(Enum):
+    UNKNOWN = "Unknown"
+    SCHEDULED = "Scheduled"
+    RUNNING = "Running"
+    COMPLETED = "Completed"
+    FAILED = "Failed"
+    ERROR = "Error"
+    CANCELED = "Canceled"
+
+
+class ExecutionMetadata(object):
+
+    avg_duration_in_seconds: int
+    max_duration_in_seconds: int
+    status: Status
+    count: int
+
+    def __init__(
+        self,
+        avg_duration_in_seconds: int,
+        max_duration_in_seconds: int,
+        status: Status,
+        count: int,
+    ):
+        self.avg_duration_in_seconds = avg_duration_in_seconds
+        self.max_duration_in_seconds = max_duration_in_seconds
+        self.status = status
+        self.count = count
+
+    @classmethod
+    def from_dict(cls, obj: Dict[str, Any]) -> 'ExecutionMetadata':
+        """
+        Construct an ExecutionMetadata from a dictionary of values.
+
+        Parameters
+        ----------
+        obj : dict
+            Dictionary of values
+
+        Returns
+        -------
+        :class:`ExecutionMetadata`
+
+        """
+        out = cls(
+            avg_duration_in_seconds=int(obj['avgDurationInSeconds']),
+            max_duration_in_seconds=int(obj['maxDurationInSeconds']),
+            status=obj['status'],
```

Comment:
Since the attribute of the class is defined as `Status`, the string in `obj['status']` should be mapped to a `Status` value. Maybe add a `from_str` class method on the `Status` class to handle this. This method could also make sure the field is defined and not blow up. This will prevent crashes if new values are added in the future.
```
    @classmethod
    def from_str(cls, s):
        try:
            return cls[s.upper()]
        except KeyError:
            return cls.UNKNOWN
```